src/Arcus.WebApi.Logging/ExceptionHandlingMiddleware.cs

```diff
@@ -62,31 +62,47 @@ public async Task Invoke(HttpContext context, ILoggerFactory loggerFactory)
             {
                 await _next(context);
             }
+            catch (Exception exception)
+            {
+                string categoryName = _getLoggingCategory() ?? String.Empty;
+                ILogger logger = loggerFactory.CreateLogger(categoryName) ?? NullLogger.Instance;
+                LogException(logger, exception);
+
+                WriteFailureToResponse(exception, context);
+            }
+        }
+
+        /// <summary>
+        /// Logs the caught <paramref name="exception"/> before the response it sent back.
+        /// </summary>
+        /// <param name="logger">The instance to track the caught <paramref name="exception"/>.</param>
+        /// <param name="exception">The caught exception during the application pipeline.</param>
+        protected virtual void LogException(ILogger logger, Exception exception)
+        {
+            logger.LogCritical(exception, exception.Message);
+        }
+
+        /// <summary>
+        /// Determine the HTTP status code based on the caught exception.
+        /// </summary>
+        /// <param name="exception">The caught exception during the application pipeline.</param>
+        /// <param name="context">The context instance for the current HTTP request.</param>
+        /// <returns>An HTTP status code that represents the <paramref name="exception"/>.</returns>
+        protected virtual void WriteFailureToResponse(Exception exception, HttpContext context)
+        {
 #if NET6_0
-            catch (Microsoft.AspNetCore.Http.BadHttpRequestException ex) 
+            if (exception is Microsoft.AspNetCore.Http.BadHttpRequestException badRequestException)
 #else
-            catch (Microsoft.AspNetCore.Server.Kestrel.Core.BadHttpRequestException ex) 
+            if (exception is Microsoft.AspNetCore.Server.Kestrel.Core.BadHttpRequestException badRequestException)
 #endif
             {
                 // Catching the `BadHttpRequestException` and using the `.StatusCode` property allows us to interact with the built-in ASP.NET components.
                 // When the Kestrel maximum request body restriction is exceeded, for example, this kind of exception is thrown.
 
-                LogException(loggerFactory, ex);
-                context.Response.StatusCode = ex.StatusCode;
+                context.Response.StatusCode = badRequestException.StatusCode;
```

Comment:
Isn't this always overwritten by line 105 ?

Reply:
Also, this (catching this BadRequestException) and retrieving the statuscode might be sometehing that is very useful, and ppl that override this default behaviour maybe want this as well.

I was thinking: maybe we can change this method, by extracting this logic outside of the virtual method and passing a 'default status code' as argument.  This default status code could then be by default 500, but if such an exception was thrown, we pass in that statuscode that is a property of the exception ?

Something like this (very roughly):

```csharp
catch( Exception ex )
{
    int defaultStatusCode = 500;

    if( exception is BadRequestException badReqException )
    {
        defaultStatusCode = badReqException.StatusCode;
    }

    WriteFailureToResponse(defaultStatusCode, ex, context);
}

protected virtual WriteFailureToResponse(int defaultHttpStatusCode, Exception ex, HttpContext context)
{
    context.Response.StatusCode = defaultHttpStatusCode;
}
````

Reply:
Yeah that's true. Think this by-default catching of bad request is kinda useful and was a bit removed by fully placing it in the overridable method. Your suggestion might be a good way to have best of both worlds, yeah. Let's try something! 👍

Reply:
Implemented something here: https://github.com/arcus-azure/arcus.webapi/pull/342/commits/3a0691f62bd4a6a9f799988e6ea1d755c5cde148

Reply:
Great!